### PR TITLE
Refactor generator base app init

### PR DESCRIPTION
**Pull request by tbukovics**
### Issue reference / description

The branch you merge from should already reference an event-formation-unit github ticket number. You can add a descriptive title, but if an issue is referenced, you don't have to.

## Checklist for submitter

- [x] Check for conflict with integration test
- [x] Unit tests pass

---

### Nominate for Group Code Review

- [x] Nominate for code review

### CODE REVIEW

File: src/generators/essudpgen/ReadoutGeneratorBase.cpp [hunk 1]
```diff
@@ -7,6 +7,7 @@
 //===----------------------------------------------------------------------===//
 // GCOVR_EXCL_START
 
+#include "Error.hpp"
```
Comment (mortenjc) [inline src/generators/essudpgen/ReadoutGeneratorBase.cpp:10]: Is this a new file?

We don't use .hpp and please remember to use <> for includes ;-)

File: src/generators/essudpgen/ReadoutGeneratorBase.h [hunk 2]
```diff
@@ -9,6 +9,7 @@
 
 #pragma once
 
+#include "App.hpp"
```
Comment (mortenjc) [inline src/generators/essudpgen/ReadoutGeneratorBase.h:12]: Where does this include come from?

Comment (mortenjc) [reply src/generators/essudpgen/ReadoutGeneratorBase.h:12]: For CLI11 stuff normally
#include <CLI/CLI.hpp>
is enough. the package is being included from conan

File: src/generators/essudpgen/ReadoutGeneratorBase.h [hunk 3]
```diff
@@ -30,7 +31,6 @@ class ReadoutGeneratorBase {
     uint64_t SpeedThrottle{0};       // 0 is fastest higher is slower
     uint64_t PktThrottle{0};         // 0 is fastest
     uint8_t headerVersion{1};        // v1 header by default
-    std::string FilePath{"none"};            // Record data file to read from
```
Comment (mortenjc) [inline src/generators/essudpgen/ReadoutGeneratorBase.h:33]: Why not keep the filename in the base class?

**Review by mortenjc — CHANGES_REQUESTED** [review]
There is some #include stuff I don't understand and a single comment about the filename

File: src/generators/essudpgen/ReadoutGeneratorBase.cpp (hunk 1 above)
Comment (tbukovics) [reply src/generators/essudpgen/ReadoutGeneratorBase.cpp:10]: This is something was sneaked in by my idea. I will clean this.

File: src/generators/essudpgen/ReadoutGeneratorBase.h (hunk 3 above)
Comment (tbukovics) [reply src/generators/essudpgen/ReadoutGeneratorBase.h:33]: So this setting in maybe not relevant to all modules. Currently biforst using it only, so I moved this setting parameter to bifrost generator.

File: src/generators/essudpgen/ReadoutGeneratorBase.h (hunk 2 above)
Comment (tbukovics) [reply src/generators/essudpgen/ReadoutGeneratorBase.h:12]: Thanks I see, I included the CLI.hpp instead.

**Review by mortenjc — APPROVED** [review]

**Pull request merged**